src/repo/xdb_repo_queryable.erl

```diff
@@ -141,7 +141,16 @@ transaction(Repo, Adapter, Fun) ->
   Opts    :: xdb_lib:keyword(),
   Res     :: {ok, any()} | {error, any()}.
 transaction(Repo, Adapter, Fun, Opts) when is_function(Fun, 0) ->
-  Adapter:transaction(Repo, Fun, Opts).
+  erlang:function_exported(Repo, prehook, 2) andalso
+    Repo:prehook(transaction, Opts),
```

Comment:
Since we are calling `Repo:prehook(transaction, Opts)`, the repo interface `xdb_repo` should expose `prehook` as callback.

Reply:
@cabol this is exactly why I'm checking if it's exported. Otherwise, how can it be overridden by a user?

Reply:
done